src/Mod/Sketcher/Gui/DrawSketchHandler.h

```diff
@@ -45,7 +45,8 @@ struct AutoConstraint
     enum TargetType
     {
         VERTEX,
-        CURVE
+        CURVE,
+        VERTEX_FOR_PRIMITIVE

```

Comment:
I would consider renaming "VERTEX_FOR_PRIMITIVE" to something showing better the intent. From the comment in the code, I get that we introduce this new enum value to support seeking for a vertex but foregoing any attempt of tangency (I did not look in deep into this detection). If that is what is intended, maybe "VERTEX_NO_TANGENCY" could be what you are looking for. Maybe I am wrong, and this mode is only useful when drawing a primitive (which is a concept from Part WB, I think, I do not remember using it in the context of the sketcher). Just, give it a thought. I won't insist on it. I did not get the naming at first and this is why I comment.

Reply:
VERTEX_NO_TANGENCY is fine for me. Done ;)